uint8 UNKNOWN=0
uint8 OPEN=1
uint8 CLOSE=2

uint8 state
